ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/UserManagementServiceConcurrentUpdates.scala

```diff
@@ -0,0 +1,164 @@
+        CreateUserRequest(Some(user1a), Nil)
+      )
+      _ = assertEquals(unsetResourceVersion(res1), CreateUserResponse(Some(user1a)))
+      user1b = User(
+        id = userId1,
+        primaryParty = "primaryParty1",
+        isDeactivated = true,
+        metadata = Some(
+          ObjectMeta(
+            resourceVersion = "",
+            annotations = Map(
+              "key1" -> "val1",
+              "key2" -> "val2",
+            ),
+          )
+        ),
+      )
+      // Update with concurrent change detection disabled
+      res2 <- ledger.userManagement.updateUser(
+        UpdateUserRequest(
+          user = Some(user1b),
+          updateMask = Some(FieldMask(paths = Seq("user"))),
+        )
+      )
+      _ = assertEquals(unsetResourceVersion(res2), UpdateUserResponse(Some(user1b)))
+      _ = assertEquals(
+        unsetResourceVersion(res2),
+        UpdateUserResponse(Some(unsetResourceVersion(user1b))),
+      )
+
+      user1aResourceVersion = res1.user.get.metadata.get.resourceVersion
+      _ = assertValidResourceVersionString(user1aResourceVersion, "newly created user")
+      user1bResourceVersion = res2.user.get.metadata.get.resourceVersion
+      _ = assertValidResourceVersionString(
+        user1bResourceVersion,
+        "updated user with concurrent change control disabled",
+      )
+
+      _ = assert(
+        user1aResourceVersion != user1bResourceVersion,
+        s"User's resource version before an update: ${user1aResourceVersion} must be different from other that after an update: ${user1bResourceVersion}",
+      )
+
+      user1c = user1b.update(_.metadata.resourceVersion := user1aResourceVersion)
+      // Update with concurrent change detection enabled, but resourceVersion is outdated
+      res3 <- ledger.userManagement
+        .updateUser(
+          UpdateUserRequest(
+            user = Some(user1c),
+            updateMask = Some(FieldMask(paths = Seq("user"))),
+          )
+        )
+        .mustFail(
+          "update a user using an out-of-date resource version for concurrent change detection"
+        )
+      _ = assertConcurrentUserUpdateDetectedError(res3)
+      user1d = user1b.update(_.metadata.resourceVersion := user1bResourceVersion)
+      // Update with concurrent change detection enabled and up-to-date resource version
+      res4 <- ledger.userManagement.updateUser(
+        UpdateUserRequest(
+          user = Some(user1d),
+          updateMask = Some(FieldMask(paths = Seq("user"))),
+        )
+      )
+      _ = assertEquals(
+        unsetResourceVersion(res4),
+        UpdateUserResponse(Some(unsetResourceVersion(user1d))),
+      )
+      user1cResourceVersion = res4.user.get.metadata.get.resourceVersion
+      _ = assert(
+        user1bResourceVersion != user1cResourceVersion,
+        s"User's resource version before an update: ${user1bResourceVersion} must be different from other that after an update: ${user1cResourceVersion}",
```

Comment:
I can't parse that sentence, consider rephrasing

Reply:
Ok, changing to ` s"User's resource versions before and after an update must be different but were the same: '$user1bResourceVersion'",`